### PR TITLE
Fix #3909

### PR DESCRIPTION
**Pull request by dushaniw**
## Purpose
Fix https://github.com/wso2/product-apim/issues/3909

## Goals
Perform validation in REST API layer when converting API Object to DTO.

## Documentation
Has Doc Impact

### CODE REVIEW

**Comment by dushaniw** [comment]
#HasDocImpact

According to the fix, the default behavior is to not expose the endpoint password unless the tenant admin specifies the config in tenant-config.json.  Please follow the below instructions to enable exposing endpoint password.

1. Sign in to the management console (https://<HostName>:9443/carbon) as admin (or tenant admin).
2. In the Main menu,  go to the Resources > Browse menu.
3. Navigate to the /_system/config/apimgt/applicationdata/ directory. 
4. Click on tenant-conf.json to load the resource in the registry browser UI and select the "Edit as text " option to edit the configurations.
5. Add the config to the existing config json as follows and click on Save Content button.
`"ExposeEndpointPassword" : true`

DocJIRA created :  https://wso2.org/jira/browse/DOCUMENTATION-8277

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java [hunk 1]
```diff
@@ -668,19 +672,67 @@ private static void setEndpointSecurityFromApiDTOToModel (APIDetailedDTO dto, AP
...
+        } catch (RegistryException e) {
+            String msg = "RegistryException thrown when getting API tenant config from registry while reading " +
+                    "ExposeEndpointPassword config";
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
+        } catch (ParseException e) {
+            String msg = "ParseException thrown when parsing API tenant config from registry while reading " +
+                    "ExposeEndpointPassword config";
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
```
Comment (a5anka) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java]: We are doing both logging and throwing Exception. Won't that lead to duplicate error logs?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java [hunk 2]
```diff
@@ -668,19 +672,67 @@ private static void setEndpointSecurityFromApiDTOToModel (APIDetailedDTO dto, AP
...
+                }
+            }
+        } catch (UserStoreException e) {
+            String msg = "UserStoreException thrown when getting API tenant config from registry while reading " +
+                    "ExposeEndpointPassword config";
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
+        } catch (RegistryException e) {
+            String msg = "RegistryException thrown when getting API tenant config from registry while reading " +
+                    "ExposeEndpointPassword config";
```
Comment (a5anka) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java]: AFAIK according to code style followed by WSO2 '+' sign should precede the line.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java (hunk 1 above)
Comment (dushaniw) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java]: yes the error is getting logged again at upper level. Thank you for pointing it out. Fixed it with https://github.com/wso2/carbon-apimgt/pull/5919/files

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java (hunk 2 above)
Comment (dushaniw) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java]: Thank you for pointing it out. Fixed it with https://github.com/wso2/carbon-apimgt/pull/5919/files

**Pull request merged**